### PR TITLE
MAINT: Replace WarningManager with catch_warnings and other numpy 1.14.0 issues

### PR DESCRIPTION
**Pull request by thequackdaddy**
I think bumpy's WarningManager has gone the way of the Monty Python parrot. 

This is causing AppVeyor to not run.

### CODE REVIEW

**Comment by thequackdaddy** [comment]
Hmmm... looks like appveyor is still unhappy. 

I think it may have something to do with the upgraded numpy... I’ll take a peek.

**Comment by josef-pkt** [comment]
Are there a lot of failures with the latest numpy 1.14?
AFAICS, there are 22 failures which are mostly/all due to an upgraded numpy.

File: statsmodels/genmod/tests/results/results_glm.py [hunk 1]
```diff
@@ -684,7 +684,9 @@ def __init__(self):
         # data set up for data not in datasets
         filename = os.path.join(os.path.dirname(os.path.abspath(__file__)),
             "stata_lbw_glm.csv")
-        data=np.recfromcsv(open(filename, 'rb'), converters={4: lambda s: s.strip(asbytes("\""))})
+        data=np.recfromcsv(open(filename, 'rb'))
```
Comment (thequackdaddy) [inline statsmodels/genmod/tests/results/results_glm.py]: Not sure why, but when you have `converters`, numpy 1.14.0 drops the column. So I think this is an easy workaround. Probably not perfect.

File: statsmodels/stats/tests/test_contingency_tables.py [hunk 2]
```diff
@@ -61,7 +61,8 @@ def test_SquareTable_from_data():
                  rslt3.summary().as_text())
 
     s = str(rslt1)
-    assert_equal(s.startswith('A 5x5 contingency table with counts:\n[[ 8.'), True)
+    assert_equal(s.startswith('A 5x5 contingency table with counts:'), True)
```
Comment (thequackdaddy) [inline statsmodels/stats/tests/test_contingency_tables.py]: numpy 1.14.0 seems to remove the leading space between the `\n[[` and the `8.`. I'm just going to check the text matches and the first element of the 2 dimensional array matches.

**Comment by thequackdaddy** [comment]
Ugh... I got the annoying `test_simulate` error... re-pushing to re-run. Sorry.

**Comment by thequackdaddy** [comment]
Hmm... there's another `rec.array` change. 

it seems you can no longer do `data[['a_field', 'b_field']].view((float, 2))`. A brief look at the docs has led me to think this is intentional design change.

My guess is that it would just be easier to use pandas for this... Otherwise, I'd have to do something like 

`np.hstack((a['a_field'], b['b_field']), dtype=float)`...

I'll play with that shortly.

**Comment by josef-pkt** [comment]
I'm giving up on our rec array usage! We have to fix something with each numpy version (or new numpy/pandas interaction for them)
If you can make any fixes more easily by switching to pandas, then you can do so already now.

I would like to drop rec array usage and support completely in 0.10, but if some changes that we need for compatibility fixes are easier by dropping them right away, then we can do so. This is especially true for changes in our "old" testing code. 
If we don't fix the code, then recarrays will only be supported for numpy != 1.4.0 or < 1.4.0, as long as we get the unit tests green.

**Comment by thequackdaddy** [comment]
Ok this looks decent to me. The coveralls fail is irrelevant. 

Thoughts?

**Comment by josef-pkt** [comment]
Looks very good to me, it's all in unit tests. Squashing would make it nicer.

@thequackdaddy  Thanks a lot for working through this

**Comment by josef-pkt** [comment]
merging

**Pull request merged**